elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/verify/DynamicConfigVerifier.java

```
@@ -0,0 +1,115 @@
+        try {
+            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            Certificate cert = keyStore.getCertificate(keyName);
+            publicKey = cert.getPublicKey();
+        } catch (KeyStoreException e) {
+            log.error(e.getMessage());
+            throw new IllegalArgumentException("Key " + keyName + " is not availabe in keystore");
+        }
+        return publicKey;
+    }
+
+    /**
+     * Verify signature of tar.gz.
+     * @param fileList
+     * @param signature
+     * @param publicKey
+     * @return whether the file can be verified by given key & signature.
+     * @throws Exception
+     */
+    public static boolean verify(String fileList, String signature, PublicKey publicKey) {
+        Signature publicSignature;
+        try {
+            publicSignature = Signature.getInstance("SHA256withRSA");
+            publicSignature.initVerify(publicKey);
+            publicSignature.update(fileList.getBytes(StandardCharsets.UTF_8));
+            byte[] signatureBytes = Base64.getDecoder().decode(signature);
+            return publicSignature.verify(signatureBytes);
+        } catch (NoSuchAlgorithmException e) {
+            log.error(e.getMessage());
+        } catch (InvalidKeyException e) {
+            log.error(e.getMessage());
+        } catch (SignatureException e) {
+            log.error(e.getMessage());
+        }
+        return false;
+    }
+
+    private static String tarContents(String archiveFile) {
+        StringBuffer sb = new StringBuffer();
+        try (TarArchiveInputStream archive = new TarArchiveInputStream(
+                new GzipCompressorInputStream(new BufferedInputStream(new FileInputStream(archiveFile))))) {
+            TarArchiveEntry entry;
+            while ((entry = archive.getNextTarEntry()) != null) {
+                sb.append(entry.getName());
+            }
+        } catch (IOException e) {
+            log.error(e.getMessage());
```

Let exception propagate as is.